src/libs/pluginprocess/pluginprocess.c

```diff
@@ -0,0 +1,379 @@
+/**
+ * @file
+ *
+ * @brief Source for the pluginprocess library
+ *
+ * Executes plugins in a separate process via fork and uses a simple
+ * communication protocol based on the dump plugin via named pipes.
+ *
+ * @copyright BSD License (see LICENSE.md or https://www.libelektra.org)
+ */
+
+#include "kdbpluginprocess.h"
+#include <kdbinvoke.h>
+// To access the plugin function pointers
+#include <kdberrors.h>
+#include <kdblogger.h>
+#include <kdbprivate.h>
+
+#include <errno.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/stat.h>
+#include <sys/types.h>
+#include <unistd.h>
+
+struct _ElektraPluginProcess
+{
+	char * commandPipe;
+	char * resultPipe;
+	int pid;
+	int counter;
+	ElektraInvokeHandle * dump;
+};
+
+static void cleanupPluginData (ElektraPluginProcess * pp)
+{
+	elektraInvokeClose (pp->dump);
+	if (pp->pid) unlink (pp->commandPipe);
+	elektraFree (pp->commandPipe);
+	if (pp->pid) unlink (pp->resultPipe);
+	elektraFree (pp->resultPipe);
+	elektraFree (pp);
+}
+
+static char * intToStr (int i)
+{
+	int size;
+	char * str;
+	size = snprintf (NULL, 0, "%d", i);
+	str = elektraMalloc (size + 1);
+	size = snprintf (str, size + 1, "%d", i);
+	return str;
+}
+
+/** Start the child process' command loop
+ *
+ * This will make the child process wait for plugin commands
+ * and execute them, returning the result to the parent. This
+ * is typically called in a plugin's open function.
+ *
+ * @param handle the plugin's handle
+ * @param pp the data structure containing the plugin's process information
+ * @see elektraPluginProcessInit how to use this function in a plugin
+ * @ingroup processplugin
+ **/
+void elektraPluginProcessStart (Plugin * handle, ElektraPluginProcess * pp)
+{
+	Key * commandPipeKey = keyNew ("/pluginprocess/commandPipe", KEY_VALUE, pp->commandPipe, KEY_END);
```

Comment:
please avoid camelCase or any other separator other than / in key names. Use `/pluginprocess/pipe/command` and `/pluginprocess/pipe/result`